internal/repositories/github/service.go

```diff
@@ -65,6 +65,13 @@ type RepositoryService interface {
 		projectID uuid.UUID,
 		providerName string,
 	) (db.Repository, error)
+	// DeleteRepositoriesByProvider deletes all repositories associated with a provider in a project
+	DeleteRepositoriesByProvider(
```

Comment:
I wonder why we need this function at all, shouldn't this just happen via an `ON DELETE CASCADE`?

Reply:
Yeah, I had that comment, then I dug in and hit the answer Ria gave.  😁

Reply:
Updated